README.md
## AppOpener 🔓

AppOpener is the python library to **open/close** any application **without knowing it's absoulute path**. The library works by making use of App Name and App Id.

Library works on python version 3.5 or above python version 3.5+

[![PyPI Downloads](https://img.shields.io/pypi/dm/AppOpener)](https://pypi.org/project/AppOpener/) &nbsp; [![PyPI Downloads](https://img.shields.io/pypi/status/AppOpener)](https://pypi.org/project/AppOpener/) &nbsp; [![PyPI Downloads](https://img.shields.io/pypi/v/AppOpener?label=AppOpener)](https://pypi.org/project/AppOpener/) &nbsp; [![Windows Only](https://img.shields.io/badge/platform-windows-blue.svg)](https://shields.io/)

<!-- > **Note**
> AppOpener is only functional on windows. -->

Visit official documentation of AppOpener [here](https://AppOpener.readthedocs.io/en/latest/).

The latest development version is always available at the [Github](https://github.com/athrvvvv/AppOpener) repository.

---

> ### Features:

1. Open applications
2. Close applicatons
3. Create list of Apps & Ids

---

### Integrating AppOpener

![](docs/img/output.gif)

See [Examples](https://github.com/athrvvvv/AppOpener/tree/module/Examples) for more.
---

> ### Install Package 📦

```
pip install AppOpener
```

> ### Quick start ⚡

``` python
from AppOpener import open, close, mklist, give_appnames
open("telegram, whatsapp") # Opens telegram and whatsapp
open("APPNAME", throw_error=True) # Raises Exception if App is not found (can be used in `close` function too)
close("telgrm", match_closest=True) # Closes telegram as "telgrm" is closest to "telegram"
mklist(name="app_data.json") # Generates an file, having key as AppName and value as AppIds.
appnames = give_appnames() # Save appnames as dictionary
```

---
> ### Building package 🔨

```
git clone https://github.com/athrvvvv/AppOpener.git
cd AppOpener
python setup.py sdist bdist_wheel
```
The latest release is always available at the Github [releases](https://github.com/athrvvvv/AppOpener/releases).

---
> ### Contributing 🤝

Contributions are welcomed. To know more visit [here](https://github.com/athrvvvv/AppOpener/blob/module/CONTRIBUTING.md).

---

### Links 🔗

- PYPI page - https://pypi.org/project/appopener/
- Official documentation - https://AppOpener.readthedocs.io/en/latest/
- Github releases - https://github.com/athrvvvv/AppOpener/releases/
- Project changelog - https://github.com/athrvvvv/AppOpener/blob/module/CHANGELOG.md/
- Issue tracker - https://github.com/athrvvvv/AppOpener/issues/

### Stay connected 🤗

- [Mail](mailto:athrvchaulkar@gmail.com)
- [Twitter](https://twitter.com/athrvvvvv)
- [YouTube](https://www.youtube.com/c/ACUNBOXING2017)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.
This changelog is addressed to PyPi module [AppOpener].

---

## [1.7] - 2023-04-30

### Added
- Ability to throw errors when an App is not found
- Made an "old_style.py" where files are created using old method if issues like #9 and #10 occur.

### Changed
- Method for finding apps in the `find ` attribute in `open` function
- Fixed `mklist` function Bug for creating file, where user has imported module

### Removed
- `['(CLS)','CLEARS SCREEN']` from `help commands`

---

## [1.6] - 2023-01-16

### Added 
- `UTF-8 ENCODING` while creating the `data.json` file ,
- Better comments for user in vs code (i.e Ex. open("whatsapp")
- Condition for platform checking (If not windows exit)
- More accurate matching for `open` and `close` functions
- Beautiful code highlighting (docs)

### Changed 
- `open_closest` and `close_closest` are now `match_closest` in `open` and `close` functions respectively.
- `git-StartApps` is now  `Get-StartApps` (So no need to maximize window to get complete Appid).
- Fix broken link to the github account (Docs).
- Fix known issue for `MKLIST` function. 
- Fix issue [#5](https://github.com/athrvvvv/AppOpener/issues/5)
- Migrated from `readthedocs` theme to sexy `material` theme.

### Removed
- Minimise the decoration made to README.md
- `pywin32` in dependencies because no need to maximize window to get complete Appids.

---

## [1.5] - 2023-01-03

### Added
- `CLOSE` function to close any application by its name.
- `open_closest=` & `close_closest=` in `OPEN` & `CLOSE` function respectively, to open/close applications which matches closest.
- See `VERSION` through `OPEN` function.
- Attribute `output=` to print context or not to print in all functions.
- `PSUTIL` dependency in `release.py`.

### Changed
- `RUN` function is been replaced by `OPEN` function

### Removed
- Attributes of `MKLIST` function
- Install AppOpener via github section in docs

---

## [1.4] - 2022-12-17

### Added
- Added pywin32 in the build spec as mentioned in https://github.com/athrvvvv/AppOpener/issues/7

### Changed
- Fixed https://github.com/athrvvvv/AppOpener/issues/6 (Now you can use AppOpener in standalone files too)

### Removed
- Install AppOpener via github section in docs

---

## [1.3] - 2022-09-04

### Added
- Added ASCII encoding to powershell command in check.py

### Changed
- space

### Removed
- "refer.txt" step (no more need to fix the encoding problems) in check.py.

---

## [1.2] - 2022-09-03

### Added
- Adding "-encoding ASCII" to the powershell command that extracts to "reference.txt" the app list

### Changed
- space

### Removed
- "refer.txt" step (no more need to fix the encoding problems).

---

## [1.1] - 2022-08-29

### Added
- Added method to fetch AppNames as keys.
- Added documentation & source links in PYPI page.

### Changed
- Fix crash while using a (.bat) files.

### Removed
- Unwanted conditions & functions.

---

## [1.0] - 2022-08-27

### Added
- Features

### Changed
- Fix many many things.

### Removed
- Unwanted conditions & functions.

[1.7]: https://pypi.org/project/AppOpener/1.7/
[1.6]: https://pypi.org/project/AppOpener/1.6/
[1.5]: https://pypi.org/project/AppOpener/1.5/
[1.4]: https://pypi.org/project/AppOpener/1.4/
[1.3]: https://pypi.org/project/AppOpener/1.3/
[1.2]: https://pypi.org/project/AppOpener/1.2/
[1.1]: https://pypi.org/project/AppOpener/1.1/
[1.0]: https://pypi.org/project/AppOpener/1.0/
[AppOpener]: https://pypi.org/project/AppOpener


release.py
from __future__ import print_function, unicode_literals, absolute_import

name = 'appopener'

branch = ''
version = '1.7'
description = "Open/Close any application by it's name."
a = open("AppOpener/README.md",encoding="utf8")
read = a.read()
long_description = read

license = 'MIT'

authors = {'Athrv' : ('Athrv Chaulkar','athrvchaulkar@gmail.com'),
            'athrvvvv' : ('athrvvvv', 'athrvchaulkar@gmail.com')}

url = 'https://github.com/athrvvvv/AppOpener/tree/module'
download_url = 'https://pypi.python.org/pypi/AppOpener'
Documentation = 'https://appopener.readthedocs.io/en/latest/'
Source = 'https://github.com/athrvvvv/AppOpener/tree/module'
Tracker = 'https://github.com/athrvvvv/AppOpener/issues'
platforms = ['Windows 10',
            'Windows 8.1',
            'Windows 11']

keywords = ['appopener',
            'open apps',
            'App Ids',
            'automation',
            'close apps']

classifiers = ['Development Status :: 5 - Production/Stable',
               'Environment :: Console',
               'Operating System :: Microsoft :: Windows',
               'License :: OSI Approved :: MIT License',
               'Programming Language :: Python :: 3.5',
               'Programming Language :: Python :: 3.6',
               'Programming Language :: Python :: 3.7',
               'Programming Language :: Python :: 3.8',
               'Programming Language :: Python :: 3.9',
               'Programming Language :: Python :: 3.10',
               'Programming Language :: Python :: 3.11'
               ]

install_requires=[
                "wheel",
                "psutil",
                "pywin32",
                "regex"
    ]
